src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetExpansionClient.cs

```diff
@@ -337,6 +389,9 @@ public virtual bool TryHandleEscape()
             {
                 ExpansionSession.EndCurrentExpansion(fLeaveCaret: 1);
                 ExpansionSession = null;
+                _symbols = default;
+                _symbol = null;
+                _arguments = _arguments.Clear();
```

Comment:
and duplication again :)

Reply:
The duplication is especially fun in the cases where it's _almost_ duplicated (but not quite)

Reply:
➡️ Tracking state containment as https://github.com/dotnet/roslyn/pull/49916#discussion_r544816693